src/ApiHandler.ts

```diff
@@ -54,8 +54,8 @@ export default class ApiHandler {
 	constructor(api: ApiPromise) {
 		this.api = api;
 		// this.wsUrl = wsUrl;
-		this.specVersion = api.createType('u32', -1);
-		this.txVersion = api.createType('u32', -1);
+		this.specVersion = api.createType('u32', 0);
```

Comment:
The problem with this is that it gets initialized to `-1` because the spec number actually could be `0`, so whenever it makes its first call, we want to force it to call `ensureMeta` and get the right metadata.